data-prepper-plugin-framework/src/main/java/org/opensearch/dataprepper/plugin/ClasspathPluginProvider.java

```diff
@@ -96,4 +98,18 @@ private void addOptionalDeprecatedPluginName(
             supportTypeToPluginTypeMap.put(supportedType, concretePluginClass);
         }
     }
+
+    private void addOptionalAlternatePluginName(
```

Comment:
Let's combine this with `addOptionalDeprecatedPluginName` as it is pretty much the same logic.

```
    private void addPossiblePluginName(
            final Map<String, Map<Class<?>, Class<?>>> pluginsMap,
            final Class<?> concretePluginClass,
            final Function<DataPrepperPlugin, String> possiblePluginNameFunction,
            final Predicate<String> possiblePluginNamePredicate) {
        final DataPrepperPlugin dataPrepperPluginAnnotation = concretePluginClass.getAnnotation(DataPrepperPlugin.class);
        final String possiblePluginName = possiblePluginNameFunction.apply(dataPrepperPluginAnnotation);
        final Class<?> supportedType = dataPrepperPluginAnnotation.pluginType();

        if (possiblePluginNamePredicate.test(possiblePluginName)) {
            final Map<Class<?>, Class<?>> supportTypeToPluginTypeMap =
                    pluginsMap.computeIfAbsent(possiblePluginName, k -> new HashMap<>());
            supportTypeToPluginTypeMap.put(supportedType, concretePluginClass);
        }
    }

```

Also, I think we can use the same logic for the main name. But, please check that is correct first!

You can use with something like this:

```
addPossiblePluginName(pluginsMap, concretePluginClass, DataPrepperPlugin::name, deprecatedPluginName -> true);
addPossiblePluginName(pluginsMap, concretePluginClass, DataPrepperPlugin::deprecatedName, deprecatedPluginName -> !deprecatedPluginName.equals(DEFAULT_DEPRECATED_NAME));
addPossiblePluginName(pluginsMap, concretePluginClass, DataPrepperPlugin::alternateName, alternatePluginName -> !alternatePluginName.equals(DEFAULT_ALTERNATE_NAME));
```

Reply:
Done